tempto-core/src/main/java/io/prestosql/tempto/internal/listeners/TestNameGroupNameMethodSelector.java

```diff
@@ -83,33 +92,29 @@ public boolean includeMethod(IMethodSelectorContext context, ITestNGMethod metho
     {
         TestMetadata testMetadata = testMetadataReader.readTestMetadata(method);
         return includeBasedOnTestName(testMetadata) &&
+                !excludeBasedOnName(testMetadata) &&
                 includeBasedOnGroups(testMetadata) &&
                 !excludeBasedOnGroups(testMetadata);
     }
 
     private boolean includeBasedOnTestName(TestMetadata testMetadata)
     {
-        if (!testNamesToRun.isPresent()) {
-            return true;
-        }
+        return testNamesToRun.map(strings -> indexOf(strings, testMetadata.testName::contains) != -1).orElse(true);
+    }
 
-        return indexOf(testNamesToRun.get(), testName -> testMetadata.testName.contains(testName)) != -1;
+    private boolean excludeBasedOnName(TestMetadata testMetadata)
+    {
+        return testNamesToExclude.filter(strings -> indexOf(strings, testMetadata.testName::contains) != -1).isPresent();
```

Comment:
substring matching is dangerous here.

eg i want to exclude `MyClass.testFoo` but i don't want to exlucde `MyClass.testFooBar`

also, we could require that the exclude comes qualified with simple class name (`MyClass.testFoo` instead of `testFoo`).

if `testMetadata.testName` is not fully qualified, then equals would be a good fit here.
otherwise
```
equals || endswith("." + exclusion)
```